test/integration/feature-flags_test.go

```diff
@@ -0,0 +1,81 @@
+//go:build integration
+
+package integration
+
+import (
+	"testing"
+
+	featureFlagsConstants "github.com/checkmarx/ast-cli/internal/constants/feature-flags"
+	commonParams "github.com/checkmarx/ast-cli/internal/params"
+	"github.com/checkmarx/ast-cli/internal/wrappers"
+	"github.com/spf13/viper"
+	"gotest.tools/assert"
+)
+
+type MockFeatureFlagsWrapper struct {
+	AllFlags     *wrappers.FeatureFlagsResponseModel
+	SpecificFlag *wrappers.FeatureFlagResponseModel
+	Err          error
+}
+
+func TestHandleFeatureFlagsIntegration(t *testing.T) {
+
+	createASTIntegrationTestCommand(t)
+	featureFlagsPath := viper.GetString(commonParams.FeatureFlagsKey)
+	featureFlagsWrapper := wrappers.NewFeatureFlagsHTTPWrapper(featureFlagsPath)
+
+	err := wrappers.HandleFeatureFlags(featureFlagsWrapper)
+	assert.NilError(t, err, "HandleFeatureFlags should not return an error")
+	assert.Assert(t, len(wrappers.FeatureFlags) > 0, "FeatureFlags map should not be empty")
+}
+
+func TestGetSpecificFeatureFlagIntegration(t *testing.T) {
+	createASTIntegrationTestCommand(t)
+	featureFlagsPath := viper.GetString(commonParams.FeatureFlagsKey)
+	featureFlagsWrapper := wrappers.NewFeatureFlagsHTTPWrapper(featureFlagsPath)
+
+	flagName := featureFlagsConstants.ByorEnabled
+	flagResponse, err := wrappers.GetSpecificFeatureFlag(featureFlagsWrapper, flagName)
+	assert.NilError(t, err, "GetSpecificFeatureFlag should not return an error")
+	assert.Equal(t, flagResponse.Name, flagName, "Feature flag name should match")
+}
+
+func TestGetSpecificFeatureFlagWithRetryIntegration(t *testing.T) {
+	createASTIntegrationTestCommand(t)
+	featureFlagsPath := viper.GetString(commonParams.FeatureFlagsKey)
+	featureFlagsWrapper := wrappers.NewFeatureFlagsHTTPWrapper(featureFlagsPath)
+
+	flagName := featureFlagsConstants.ByorEnabled
+	flagResponse, err := wrappers.GetSpecificFeatureFlag(featureFlagsWrapper, flagName)
+	assert.NilError(t, err, "GetSpecificFeatureFlagWithRetry should not return an error")
+	assert.Equal(t, flagResponse.Name, flagName, "Feature flag name should match")
```

Comment:
what the difference between the test above and this one?

Reply:
update